### ISSUE TITLE
News - does not work

### ISSUE THREAD

**Issue by MattiasJay** [Skjutsgruppen/skjutsgruppen-reactnative#581]
Hi @ruchidhami , the "news" feature in the admin panel does not work. I tried to publish a new news item now, and nothing happened. Not published.

Other questions regarding news: 

1. What's the max number of characters that can fit in a news title?
2. The box where we are to write the description, in the admin panel, is very small.
3. We talked on our meetings @anjhero and @anjesh that news should not only be limited by time. So the "From" and "Until" should also be able to be set to "unlimited" or "forever".
4. How do we delete or edit news?
5. Does links work (as specified in our specifications document). Was not able to test this.

**Comment by ruchidhami** [comment]
Hello @MattiasJay, It's strange, I'v added news and it has been successfully published. From which browser you have tried?

1. What's the max number of characters that can fit in a news title?
     > There is not such limitation for news title for now. If needed, please suggest we will implement so .

2. The box where we are to write the description, in the admin panel, is very small.
    > Ok. We will implement. Or you can just provide us the size.

3. We talked on our meetings @anjhero and @anjesh that news should not only be limited by time. So the "From" and "Until" should also be able to be set to "unlimited" or "forever".
   > Ok, make sense. If the news will be forever then the From and Until should be optional and if it for the certain time then we can simply select the date. 

4. How do we delete or edit news?
   > After adding the news there is a section of news listing from where we can simply delete and edit news.

5. Does links work (as specified in our specifications document). Was not able to test this.
   > Sorry. I didn't get your point . Could you please clarify ?

**Comment by MattiasJay** [comment]
Hi, I've tested on Chrome. Now I tried it again on Chrome and then on Safari (both on Mac), and still nothing happens :-)

What browser have you published from @ruchidhami ?

Answers:

2: Feel free with the size, just make it larger (like the box I'm writing in here on Github)

5: IF a link is provided in the news text (description) it should become blue and open in the default browser of the phone.

**Comment by ruchidhami** [comment]
Hello @MattiasJay.   Today I've added news from the Chrome(on Mac) and it works fine. How much time it takes to load ? Does it throw any error?

**Comment by MattiasJay** [comment]
Hi I tried now again, and still nothing. Which Chrome version are you using @ruchidhami ?

When I press "Submit" the page just reloads and there is no error message. Just the same page in admin reloaded and nothing pulbished. I then go back to "News" in the menu and there's a red box above the "Add News" button saying "object Object" in brackets.

Can you see that we are online at the same time on Slack and do a live test where you publish a news item live in the app, so I can see it?

**Comment by MattiasJay** [comment]
Hi again @ruchidhami , did you see the above comment? I've now tested it on a mobile device - latest Chrome on iOS - and got this error when trying to publish news: "413 Request Enity Too Large" and "nginx/1.15.8"

**Comment by MattiasJay** [comment]
Hi again @ruchidhami ! Can we trie to test this during Wednesday?

> Can you see that we are online at the same time on Slack and do a live test where you publish a news item live in the app, so I can see it?

**Comment by ruchidhami** [comment]
Hello @MattiasJay , I've research on **"413 Request Enity Too Large" and "nginx/1.15.8"**  and got to know that , if the image size is bigger then such type of error will occurs. Please make to sure that image size is not greater than 500kb. Can you please try to post the news with less image size?

**Comment by MattiasJay** [comment]
Backlog: chatting with you about this right now on Slack @ruchidhami :-)

**Comment by MattiasJay** [comment]
Backlog: @ruchidhami and me today did first real news post. It still doesn’t work from my Chrome browser, so Ruchi posted it from her computer. This is the result of the first try. A still from the short film I did of it:

![GitHawk Upload by MattiasJay](https://i.imgur.com/saoklGU.jpg)

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by MattiasJay** [comment]
This is what me and @ruchidhami concluded on Slack:

1) Can't open the news post in the app (the blinking in the movie version of above is me trying to open it)
2) This resolution will not do as you can see 😄  . We will be using this for ads and we need much higher resolution for this. 
3) Also, we need some kind of direction on how the photo will be stretched out and zoomed in. As you can see this is very zoomed in.
4) Regarding the title of the news, it's of from the design.

For reference:
- Design for long title: https://projects.invisionapp.com/d/main#/console/12489615/280148483/preview
- Design for short title: https://projects.invisionapp.com/d/main#/console/12489615/280148587/preview

5) Things that I were unable to test since the news was not able to open:
- formatting, is there formatting? (just like there is in comments in rides, I suspect there is)
- links, does links work? (see above in the first post of this issue)

I'm happy to see that we are now getting closer to this. So keep up the good work!

**Comment by RaiSaugat** [comment]
@MattiasJay Are those two design final for the news card? The main reason behind to blur the background image was to make the text readable.  Please, give your suggestions on how to implement the design. :)

**Comment by anjesh** [comment]
@MattiasJay we need to have simpler solution here. In design we see comments and there's still not detail page yet and we are not sure how to go about that. We need to get things ready with simple solution. 

Here's our suggestion

Admin may add the following
* Title (compulsory)
* Description or URL
* Picture (compulsory)
* From-to dates

If the admin adds url, then the feed news image will open the link mentioned in the URL field. The text image will be shown without blurred image so if the text will not be readable.

If the admin adds description, the the feed news image will open a new page with the description. The text image will be shown without blurred image so if the text will not be readable.

There we can't have comments at this moment because that will require change in the database. Our suggestion is to remove both share and comments from the news . These things and any other remaining could be moved for future enhancement.

Please let us know what you think before we start working on this. 

cc @ruchidhami

**Comment by MattiasJay** [comment]
Hi @RaiSaugat and @anjesh . Thanks for your questions! It seems like this has not been communicated from the weekly meetings? We have in those meetings concluded that we will not have comments in news and that news is basically the same design as "Experience". 

We also said that, if needed, we can trie the "darkening" of images (greyish overlay?) we use for group images for the feed version of image in news. Remeber, @anjhero ? (Anjan have sent me a couple of different suggestions on how this may look). 

To make this clearer and faster, this PSD is now on the Drive:

![news](https://user-images.githubusercontent.com/10048880/57921432-9ebe9700-789d-11e9-9c66-751110a7c36a.jpg)

So, we seem to be on the same page @anjesh , great! 

1) Opening URL from feed directly in web browser sounds good like a good option to have! Thanks for this!

2) I also suggest the ability to not have a title, since this may look good when/if going directly from feed to opening link in browser (and in some cases for ad use of this functionality).

3) Lastly, just to be clear, we will use this function for ads. So blurred will not work (companies would not like their images/logos blurred). I suggest we instead opt for a set size of the pic. I've made open view and feed view basically the same size. Can you trie the "darkening" @anjhero suggested before?

Good job everyone! You are great!

@ruchidhami

**Comment by anjesh** [comment]
wonderful. 

We will have the following

* Admin may add the following fields
  * Title
  * Description (will have basic formatting option)
  * URL 
  * Picture (compulsory)
  * From-to dates

Only the picture is compulsory. If the admin puts the URL, it will ignore the description part (if anything written) and opens the link in browser. If there's no URL, then the image will open the new view with description. If title is present, then title will be shown above picture as you have asked. It's the responsibility of the image uploader to enhance image to ensure readability. 

For now, we will not add the `share` and `more news` option. Those will be covered in future updates. 

Please confirm before we start working on this. @MattiasJay

**Comment by MattiasJay** [comment]
I love the progress on this! Great @anjesh !

When you say 
> For now, we will not add the share and more news option. Those will be covered in future updates.

- Do you mean _you_ will update this within the scope of our agreement?

We can meet you on this and be ok with this 👍 . Thank you @anjesh !

We do need in-text link to work with this update. That is core to the news function. Look at the blue link example in the picture above (in the text it says "https://example.link).

This is an example of how the in-text link works:

1. Go to "The Self-Sustaining Garden" in the app 
2. Scroll down to "Help translate the app"
3. _IF_ link is clicked THEN it opens in the default browser of the phone.

So, like this:

**Admin have to add the following fields:**
- Picture (compulsory). Picture is at least the same quality resolution we have for all other photos. 
- The picture will ha a set size, similar to like we do with group photos. Please provide this size as a text in admin section so we remember this.
- _IF_ Title on top of picture _THEN_ make sure it's visible. How about the darkening we use for groups @ruchidhami @RaiSaugat ?

**Admin may add the following fields:**
- Title
- Description (will have basic formatting option, like line break)
- URL in the description text, _IF_ clicked _THEN_ opens a in default browser of phone
- From-to dates. _IF_ not added _THEN_ the news is forever

**Extra function:**
- IF no description THEN the news item opens directly in default browser of the phone.

**Automatic addition from the system:**
- When the news was uploaded (see example in the design)

**Admin:**
- We still have to make sure this works on Chrome.
- Edit news
- Delete news

---

To be clear; the extra functionality is a bonus for us but not needed:
> **Extra function**
> - IF no description THEN the news item opens directly in default browser of the phone.

Great job everyone! Looking forward to see this come into reality this week. You rock!

**Comment by anjesh** [comment]
> Do you mean you will update this within the scope of our agreement?

No. This will not be done within the scope of this agreement. 

> We do need in-text link to work with this update. That is core to the news function. Look at the blue link example in the picture above (in the text it says "https://example.link).

That's what we mean by basic formatting. You will be able to add link in the description text. There will be editor with basic features like adding link, maybe bold, italic. 

> Extra function:
> IF no description THEN the news item opens directly in default browser of the phone.

This doesn't sound like extra function but the feature that we will need in this scope. URL (opening external browser) will take precedence over opening to new view with description. 

> We still have to make sure this works on Chrome.
> Edit news
> Delete news

Yes these will be covered.

> Automatic addition from the system:
> When the news was uploaded (see example in the design)

Not clear what you mean here. When the news is uploaded, it appears in the feed as it is now. Is that what you mean?

**Comment by MattiasJay** [comment]
> That's what we mean by basic formatting. You will be able to add link in the description text. There will be editor with basic features like adding link, maybe bold, italic.

Ah good! I thought you meant that IF any link in _the description_, it would disregard the whole description and send to default browser. I'm happy that this is not what you meant :) . So what you mean is an extra field in the admin section with _only_ link.

>> Automatic addition from the system:
>> When the news was uploaded (see example in the design)

> Not clear what you mean here. When the news is uploaded, it appears in the feed as it is now. Is that what you mean?

I'm referring to the date the news was added that in the design above is the small small line of text saying "2019-02-03"

> No. This will not be done within the scope of this agreement.

Ok. Add a new issue with these features so we have it in the backlog.

**Comment by anjesh** [comment]
Thanks.

> I'm referring to the date the news was added that in the design above is the small small line of text saying "2019-02-03"

Understood now.

> Ok. Add a new issue with these features so we have it in the backlog.

Added https://github.com/Skjutsgruppen/skjutsgruppen-reactnative/issues/587

**Comment by MattiasJay** [comment]
Can you give me a progress update on this, @ruchidhami @RaiSaugat & @anjesh ?

**Comment by ruchidhami** [comment]
Hi! @MattiasJay  This has been implemented . Please confirm it from test flight.

**Comment by MattiasJay** [comment]
Hi @ruchidhami , I'm testing this right now. Still not possible to upload news from Chrome so I've downloaded Firefox instead. Same thing there, nothing happens. After pressing "submit" the page reloads --> same page again --> nothing has been published in the feed in the app. This is now very urgent for us. @anjesh 

**Update:**
We have now tested on a PC as well, with the same result.

So far these results:
- Chrome on Mac, nothing happens (see above)
- Firefox on Mac, nothing happens (see above)
- Chrome on iPhone6, first gets whole page error about picture too large. Then uses smaller picture and nothing happens, no error but nothing happens.
- Chrome on PC, first gets whole page error about picture too large. Then uses smaller picture and nothing happens, no error but nothing happens.

**Comment by ruchidhami** [comment]
Hello @MattiasJay , Could you please provide me the news details, so that I can try to publish from my pc? Now I've set upto 20MB image to upload.

**Comment by MattiasJay** [comment]
I see the update today with the editing tool, looks good @ruchidhami .

Backlog: we are still not able to publish from Chrome on Mac. But for design purposes @ruchidhami has tested on her computer. Here are the feedback on the first test:

![01_comments_on_news_design](https://user-images.githubusercontent.com/10048880/58865401-ca6fb880-86b6-11e9-88cd-4cc7def0172e.jpg)

- [x] Spacing

- [x] Size of box

- [x] Also, the "NEWS" in the upper right corner should be with large letters in both Swedish and English. In Swedish: "NYHETER", in English: "NEWS"

**Comment by MattiasJay** [comment]
![02_comments_on_news_design](https://user-images.githubusercontent.com/10048880/58867906-8fbc4f00-86bb-11e9-9e50-19aae0a74f31.jpg)

- [x] Shorter picture

- [x] Similiar feedback as above: NEWS and NYHETER (large letters)

- [x] No drop shadow on "NYHETER" and "NEWS"

- [x] Correct the back arrow

- [x] Spacing of title

**Comment by MattiasJay** [comment]
![03_comments_on_news_design](https://user-images.githubusercontent.com/10048880/58868594-f130ed80-86bc-11e9-89a9-d8a93f7a8479.jpg)

- [x] Implemented top box design priciple (the back arrow should not be floating on top of content)

- [x] Correct space between text and title

- [x] Correct spacing in text

- [x] Correct font for text

- [x] Bold link 

- [x] Correct color for link

- [x] Correct font and size for date

- [x] Correct design for date and line relationship

- [x] General correct spacing of the text (it should not flow to the edge of the screen). See similar issue here: https://github.com/Skjutsgruppen/skjutsgruppen-reactnative/issues/566#issuecomment-498059309

**Comment by MattiasJay** [comment]
Tried with a long title, that broke the design.

- [x] Set a character limit for the title, so it doesn't break design. 

- [x] Add a text with the optimal picture size in the admin panel:

- Hight and length
- Maximum file size
- Optimal / suggested type pf picture (jpg or png or other)

**Comment by RaiSaugat** [comment]
@MattiasJay  We implemented the news screen on feed according to the 
https://projects.invisionapp.com/d/main#/console/12489615/280148483/preview . But we didn't have the psd file so the design didn't match.
To match exactly with design, could you provide psd file for the news feed on feed screen.
Also please provide:
1) Character limit for the title.
2) Maximum size of the picture

And what does height and length refers to ? Please reply as soon as possible.

**Comment by ruchidhami** [comment]
Hello @MattiasJay,

> Maximum Image file size
  
     Upto 20 MB.

> Optimal / suggested type of picture (jpg or png or other)

    Only JPG, JPEG and PNG are allowed.

**Comment by MattiasJay** [comment]
Hi @RaiSaugat and @ruchidhami ! Thanks for your swift and good work on this! We appreciate it a lot!

> To match exactly with design, could you provide psd file for the news feed on feed screen.

These designs are now on the Drive as well.

> 1. Character limit for the title.

In accordance with the design, the character limit should be `50 characters`

Is the "Maximum size of picture" the same as @ruchidhami then answered to?  (20 MB?)

The optimal length and height is something that social media platforms use to know how to cut images for optimal view, for instance in Photoshop. Here are examples from larger social media platforms, example: "Facebook cover image size: 800 x 312 px" - https://www.brandwatch.com/blog/social-media-image-sizes-guide/

It's like the example you gave me yesterday @ruchidhami for group images: _"first image will crop  by 700 x 600 pixels"_ (quote from Slack)

You're doing a good job! Anything more you need to proceed?

**Comment by MattiasJay** [comment]
Hi again @ruchidhami and @RaiSaugat !

Nice update today! We like what we're seeing, good job! 👍 

It seems like we are very close now. While testing, some things are still to be done. To be able to have quick overview, please check the check boxes above of what is done. Thanks!

1) When clicking a link _in text_ and then returning to the app, the link is suddenly embedded in the app:

![GitHawk Upload by MattiasJay](https://i.imgur.com/jI7fvyp.jpg)

- [x] Fixed so link does not embed itself in the app

2) Another color appears, see where the arrow is pointing

![GitHawk Upload by MattiasJay](https://i.imgur.com/gx2oU7S.jpg)

- [x] Made background white

3) Add more white space beneath the date. Look at the orginial design, even if we don't have the share button or "more news". See the white space between the date and the share button? Add that space.

![GitHawk Upload by MattiasJay](https://i.imgur.com/swcqLIh.jpg)

4) Currently hard to tell if picture quality is just bad because of the resizing the app makes or something else. Have you looked in to the above question of optimal higth and lenth @ruchidhami ?

**Comment by MattiasJay** [comment]
Hi again @ruchidhami and @RaiSaugat , we need an update about this. Please use the checkboxes above. We tried another test now and there's cropping of the image, but what is the cropping you use? (same question as above; optimal size)

**Comment by MattiasJay** [comment]
While waiting for your answer we have begun experimenting. We tried to upload this news picture:

![test_appen](https://user-images.githubusercontent.com/10048880/59685075-9da1c200-91db-11e9-8372-3afe6e3f535d.jpg)

But this is the result of the cropping in the app:

![GitHawk Upload by MattiasJay](https://i.imgur.com/iklw3ri.jpg)

I will be calling you about this tomorrow @anjesh

**Comment by anjesh** [comment]
I think there has been confusion about the image spec and the screen and with this comment https://github.com/Skjutsgruppen/skjutsgruppen-reactnative/issues/581#issuecomment-498595353, things got confusing because the image and design appears to differ according to devices. We either have to crop the image accordingly and it makes things worse because cropping feature doesn't have any context on which part to crop and which part to show. 

Our suggestion is not to do any cropping and show the image as it is and it is up to the user to maintain the dimension of the image appropriately like it should maintain the aspect ratio of the card as per your design. In case the ratio doesn't match, white space will show. So if you upload the image with larger width, then it's height will be small. There will be white space on top and bottom of the card.

cc @ruchidhami @umuieme

**Comment by MattiasJay** [comment]
Hi! We have since a year had a good cropping of images connected to groups. For this it's adjusted accordingly. So there seems to be working solutions for this already implemented in our system.

I assume there's a different device solution that apps like Facebook use, have you looked into how they do it and what are your conclusion after this?

~~A big thing is that quality of the picture should remain. This does not seem to be the case right now @anjesh~~.  

Is any update of the apps needed or is this a backend thing that we can implement right now today?

**Comment by MattiasJay** [comment]
Scratch the part above with the picture quality @ruchidhami and @anjesh . It seems to be working good now for the purpose of the ad we are publishing today.

**Comment by MattiasJay** [comment]
Update: In conversation with @umuieme on Slack we have decided to keep the size of the news function photos they way it is in version 1.3.1. This will do for now.

**Comment by MattiasJay** [comment]
We have an breaking of words due to screen sizes that seems to happen on mainly Android phones. Here are two examples from @umuieme 's Android and our Android. It's the word "tillsammans" that breaks up in both instances. Any suggestions on how to fix this?

Example one:
![GitHawk Upload by MattiasJay](https://i.imgur.com/nGxj8zv.jpg)

Example two:
![Screenshot_2019-06-21-16-13-51](https://user-images.githubusercontent.com/10048880/59917308-eba80700-9421-11e9-827f-2f1586c20965.png)

**Comment by ruchidhami** [comment]
@MattiasJay 
The below screenshot has taken from the android phone. After talking with team, we have concluded that,  Word breaks will definitely happens at some point, it depends on the size of the screen. We can't control the word break. 
 
Suggestion: We can make font size smaller, but we're not sure it will works for all screen size.

> ![IMG_20190621_173139](https://user-images.githubusercontent.com/13533382/59920595-b06dfe00-944a-11e9-91a3-809d027f9832.jpg)

**Comment by MattiasJay** [comment]
We can use @umuieme 's phone as an example. Trie to just implement an Android fix for this, using @umuieme 's phone as blue print. It seems like we just need to make sure the "s" in "tillsammans" appears on the same line on his phone, so not a big change in font size, just a couple of pixels @ruchidhami .

Screenshot his phone when you have done that and upload here so we can see. Thanks for looking into this @ruchidhami !
